universum/modules/structure_handler.py

```diff
@@ -264,6 +265,20 @@ def execute_steps_recursively(self, parent: Step,
 
         return not some_step_failed
 
+
+    def execute_conditional_step(self, step, step_executor):
+        self.configs_current_number += 1
+        step_name = self._build_step_name(step.name)
+        conditional_step_succeeded = False
+        with self.block(block_name=step_name, pass_errors=True):
+            process = self.execute_one_step(step, step_executor);
+            conditional_step_succeeded = not process.get_error()
+        step_to_execute = step.if_succeeded if conditional_step_succeeded else step.if_failed
```

Comment:
It will be handled as a separate work item ("check work with only one branch step specified") under the general [task](https://github.com/Samsung/Universum/issues/709).